core/src/edu/cornell/gdiac/temporary/GDXRoot.java

```diff
@@ -144,15 +150,34 @@ public void exitScreen(Screen screen, int exitCode) {
 			setScreen(menu);
 			loading.dispose();
 			loading = null;
-		} else if (exitCode == ExitCode.TO_PLAYING) {
-			screen.hide();
+		}
+		// TODO: @Melanie, modify this condition to include the playing
+		// TODO: also add a new else-if branch condition for exit code to the level screen
+		else if (exitCode == ExitCode.TO_PLAYING) {
+//			screen.hide();
+			// TODO: probably add the level populate here
+//			to playing is now to level select
+			levelscreen.setScreenListener(this);
+			System.out.println("to level again");
+			levelscreen.populate(directory);
+			setScreen(levelscreen);
+			levelscreen.show();
+
+
+		} else if (exitCode ==ExitCode.TO_LEVEL){
+//			to level is right now to game
+
+
 			playing.setScreenListener(this);
-			playing.readLevel(directory);
+			String fileName = levelscreen.getSelectedJson();
+//			System.out.println(fileName);
+			playing.readLevel(directory,fileName);
 			playing.populate(directory);
 			playing.initializeOffset(calibration.getOffset());
 			setScreen(playing);
 			playing.show();
-		} else if (exitCode == ExitCode.TO_EDITOR) {
+		}
```

Comment:
I'm not sure which one this is, but I'm assuming TO_LEVEL is actually to the game? I would at the very least rename them and recomment in ExitCode

Reply:
yeah the two are swapped at the moment.

Reply:
ok this is changed